synapse/lib/auth.py

```diff
@@ -1286,10 +1287,16 @@ async def setAdmin(self, admin, gateiden=None, logged=True):
     async def setLocked(self, locked, logged=True):
         if not isinstance(locked, bool):
             raise s_exc.BadArg(mesg='setLocked requires a boolean')
+
         if logged:
             await self.auth.setUserInfo(self.iden, 'locked', locked)
+            if not locked:
+                await self.auth.setUserInfo(self.iden, 'policy:attempts', 0)
```

Comment:
looking at this in the context of visi's previous comment; should we only be clearing `policy:attempts` when the `policy:attempts` property is set / > 0 ?

Reply:
Updated to only clear the attempt counter if the policy includes it.

Reply:
If they were to change the policy while a user is locked out, we still want it cleared right?

Reply:
( think cisphyx's suggestion of clearing if it if's not None or 0 makes the most sense )